web-common-plutus/src/ValueEditor.purs

```diff
@@ -74,9 +75,15 @@ balanceRow handler currencyIndex currencySymbol tokenIndex (Tuple tokenName amou
                       , placeholder "Amount"
                       , min zero
                       , onValueInput
-                          $ \str -> do
-                              newAmount <- BigInteger.fromString str
-                              pure $ handler $ SetBalance currencySymbol tokenName newAmount
+                          $ \str ->
+                              -- default to 0 in case of empty or invalid input
+                              -- (for reasons I have yet to fathom, this doesn't work when you delete "0";
+                              -- until I get to the bottom of that, this is at least an improvement)
+                              let
+                                newAmount = fromMaybe zero $ BigInteger.fromString str
+                              in
+                                do
+                                  pure $ handler $ SetBalance currencySymbol tokenName newAmount
```

Comment:
Oh, small thing, but this `do` is unnecessary. (You only need it if there's more than one expression in the block.)